src/wrapper/SerializationWrapper.cpp

```diff
@@ -67,6 +67,70 @@ char fs_serializer_openmode(void* serializer)
     return ' ';
 }
 
+int fs_serializer_metainfo(void* serializer)
+{
+    return reinterpret_cast<const Serializer*>(serializer)->globalMetainfo().size();
+}
+
+void fs_serializer_metainfo_key_lengths(void* serializer, int* lengths)
```

Comment:
The question is whether we want to provide a C interface to our library, which is then used by Fortran, Python, ... or if we prefer writing a Python library in C, using boost::python or whatever we might want to have.

In the former case, which is our current strategy, no, there is no way of asserting away out-of-bound cases or fix them. If we want to go for a Python extension written in C/C++, things would change completely. A lot of things would be much easier implemented in C++ than the current C/Python implementation. On the other side that would require more dependencies.

What you suggest is to change the design itself of the interoperation, isn't it?